SwanCustomEnvironments/swancustomenvironments/scripts/makenv.sh

```diff
@@ -0,0 +1,292 @@
+            shift
+            shift
+            ;;
+        --clear|-c)
+            CLEAR_ENV=--clear
+            shift
+            ;;
+        --help|-h)
+            print_help
+            exit 0
+            ;;
+        --accpy)
+            ACCPY_CUSTOM_VERSION=$2
+            shift
+            shift
+            ;;
+        *)
+            >&2 _log "ERROR: Invalid argument: $1" && _log
+            print_help
+            exit 1
+            ;;
+    esac
+done
+
+# --------------------------------------------------------------------------------------------
+
+ENV_PATH="/home/$USER/${ENV_NAME}"
+
+# Checks if a name for the environment is given
+if [ -z "$ENV_NAME" ]; then
+    >&2 _log "ERROR: No virtual environment name provided." && _log
+    print_help
+    exit 1
+fi
+
+# Checks if an environment with the same name was already created, if --clear is not passed
+if [ -d "$ENV_PATH" ] && [ -z "$CLEAR_ENV" ]; then
+    >&2 _log "ERROR: Virtual environment already exists."
+    exit 1
+fi
+
+# Checks if the provided Acc-Py version is valid
+if [ -n "$ACCPY_CUSTOM_VERSION" ] && [ ! -e "$ACCPY_BASE/$ACCPY_CUSTOM_VERSION" ]; then
+    >&2 _log "ERROR: Invalid Acc-Py version. Options: ${ACCPY_ALL_VERSIONS_STR}"
+    exit 1
+fi
+
+# Checks if a requirements file is given
+if [ -z "$REQUIREMENTS" ]; then
+    >&2 _log "ERROR: No requirements provided." && _log
+    print_help
+    exit 1
+# Checks if the provided requirements source is found
+elif [ -f $REQUIREMENTS ]; then
+    if [[ ${REQUIREMENTS##*.} != "txt" ]]; then
+        >&2 _log "ERROR: Invalid requirements file."
+        exit 1
+    fi
+    REQ_PATH=$REQUIREMENTS
+elif [[ $REQUIREMENTS == http* ]]; then
+    # Extract the repository name from the URL
+    repo_name=$(basename $REQUIREMENTS)
+    repo_name=${repo_name%.*}
+    
+    mkdir -p $HOME/SWAN_projects
+    REPO_PATH=$HOME/SWAN_projects/${repo_name}
+
+    # Checks if the repository already exists in the home directory
+    if [ -d "${REPO_PATH}" ]; then
+        if [ -z "$CLEAR_ENV" ]; then
+            >&2 _log "ERROR: ${REPO_PATH} already exists. Use --clear to reclone the repository."
+            exit 1
+        else
+            rm -rf ${REPO_PATH}
+        fi
+    fi
+
+    echo "Cloning the repository from ${REQUIREMENTS}..."
+    # Clone the repository
+    git clone $REQUIREMENTS -q --template /usr/share/git-core/templates "${REPO_PATH}" || { >&2 _log "ERROR: Failed to clone repository"; exit 1; }
+
+    REQ_PATH=${REPO_PATH}/requirements.txt
+    # Check if requirements.txt exists in the repository
+    if [ ! -f "${REQ_PATH}" ]; then
+        rm -rf ${REPO_PATH}
+        >&2 _log "ERROR: requirements.txt not found in the repository"
+        exit 1
+    fi
+else
+    >&2 _log "ERROR: Requirements not found (${REQUIREMENTS})."
+    exit 1
+fi
+
+# Checks if the requirements file is not empty
+if [ ! -s "$REQ_PATH" ]; then
+    >&2 echo "ERROR: Requirements file is empty."
+    exit 1
+fi
+
```

Comment:
Is this an error or we just create an empty environment?

Reply:
In that case, I think it would make more sense to just allow the requirements to be an optional field

Reply:
Requirements will now be a repository (EOS path or git repo URL). If I understand correctly, here you want to cover the case in which the requirements.txt exists inside that repository, but it's empty. What I'm saying is that, in that case, you don't error out, you just have the same behaviour as a pip install requirements.txt with a file that is empty.